rules/CodeQuality/Rector/Switch_/SwitchTrueToIfRector.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\Switch_;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Stmt\Case_;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\Node\Stmt\Switch_;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\CodeQuality\Rector\Switch_\SwitchTrueToIfRector\SwitchTrueToIfRectorTest
+ */
+final class SwitchTrueToIfRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Change switch (true) to if statements', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        switch (true) {
+            case $value === 0:
+                return 'no';
+            case $value === 1:
+                return 'yes';
+            case $value === 2:
+                return 'maybe';
+        };
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        if ($value === 0) {
+            return 'no';
+        }
+
+        if ($value === 1) {
+            return 'yes';
+        }
+
+        if ($value === 2) {
+            return 'maybe';
+        }
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Switch_::class];
+    }
+
+    /**
+     * @param Switch_ $node
+     * @return Stmt[]|null
+     */
+    public function refactor(Node $node): ?array
+    {
+        if (! $this->valueResolver->isTrue($node->cond)) {
+            return null;
+        }
+
+        $newStmts = [];
+
+        $defaultCase = null;
+
+        foreach ($node->cases as $case) {
+            if (! $case->cond instanceof Expr) {
+                $defaultCase = $case;
+                continue;
+            }
+
+            $if = new If_($case->cond);
+            $if->stmts = $case->stmts;
+
+            $newStmts[] = $if;
+        }
+
+        if ($defaultCase instanceof Case_) {
+            return array_merge($newStmts, $defaultCase->stmts);
+        }
+
+        return $newStmts;
```

Comment:
when `$newStmts` is empty array, it should be skipped, as it valid code, ref https://3v4l.org/qGdQ0 and it will got error as return empty array of node.

Reply:
https://github.com/rectorphp/rector-src/pull/3556